include/math_utils.h

```diff
@@ -319,5 +319,11 @@ template float remap<float>(const float in_min, const float in_max,
 template double remap<double>(const double in_min, const double in_max,
                               const double out_min, const double out_max,
                               const double v);
-							
+
+// From: http://graphics.stanford.edu/~seander/bithacks.html#DetermineIfPowerOf2
+constexpr bool is_power_of_two(int v)
+{
+	return v && ((v & (v - 1)) == 0);
+}
```

Comment:
Yeah, I just wanted to have something that works at compile-time. I'm spoiled by Nim where you can use a significant part of the standard library at runtime, including all math functions, file I/O, etc.

Your solution is cleaner, so changing to that one.

Reply:
Okay, unfortunately you suggestion doesn't seem to work:

<img width="1064" alt="image" src="https://github.com/dosbox-staging/dosbox-staging/assets/698770/90dadef4-d67d-4d01-935b-a0c205f28daa">

Using your first suggestion then 🤷🏻

Reply:
Yeah, I just tried compiling it and also get a compile error.  What C++ version are we on?  It says the count method is consexpr since C++ 23.  I suppose we can't use this if we're on an older version.

https://en.cppreference.com/w/cpp/utility/bitset/count

Reply:
> What C++ version are we on?

On C++20, but not all C++20 features are supported in all the compilers that we use. So we use as much as we can.

We're not using C++23 features (and shouldn't as it's too early for that).

Reply:
Actually, found a better solution.  This one compiles for me and is simpler (named after the x86 instruction and in standard since C++20)

```cpp
constexpr bool is_power_of_two(unsigned int v)
{
	return std::popcount(v) == 1;
}
```

Reply:
Re-opening so you can see my above comment.  Up to you if you don't want to use it though.

Reply:
Very nice find @weirddan455, I like it! Simpler == better, so changing.

Reply:
Just noticed this, what about `std::has_single_bit()`?

Reply:
Cheers, I can try that quickly. Hopefully it will be accepted by all our compiler targets 🤞🏻

Reply:
Welcome to C++ where the standard has 5 different ways to do the same thing :wink:

Reply:
@weirddan455 

"C++ is like an octopus made by nailing extra legs onto a dog."

"The 11 in C++11 refers to the number of legs that have now been nailed onto the dog whilst attempting to build a better octopus."

Still beats Cobol, PHP, and JavaScript... although that doesn't say much 😎